### PR TITLE
Update templates to incorporate net7 changes

### LINKED ISSUE #41893: Update project templates to leverage new authentication APIs
Update the project templates in response to #39855 when it's completed

### PR DESCRIPTION
**Pull request by captainsafia**
## Description

- Removed `UseAuthentication` and `UseAuthorization` middleware registration from templates
- Update `Swashbuckle.AspNetCore` version to 6.4.0
- Set accelerator key for "Do not use top-level statement option" in Visual Studio

Closes https://github.com/dotnet/aspnetcore/issues/41893 and https://github.com/dotnet/aspnetcore/issues/42719

## Customer Impact

Over the past few previews, we've been iterating on a feature to automatically register authentication and authorization middlewares when a user has configured authentication schemes or authorization policies. This PR removes the manual registration of these middlewares from the templates and contributes to our overall goal of reducing the amount of boilerplate that users have to write in order to configure authentication.

This PR also updates the `Swashbuckle.AspNetCore` dependency to incorporate code changes that light up the new `WithOpenApi` APIs shipping in preview5.

## Regression?

- [ ] Yes
- [X] No

## Risk

- [ ] High
- [ ] Medium
- [X] Low

Low risk because:

- Change is limited to ASP.NET templates with authentication configured.
- Changes require minimal reactions in identity scaffolding.

## Verification

- [X] Manual (required)
- [X] Automated

## Packaging changes reviewed?

- [ ] Yes
- [ ] No
- [X] N/A

### CODE REVIEW

**Comment by dougbu** [comment]
@captainsafia aren't these changes incomplete w/o removing some of the parameters from template.json files❔ We shouldn't leave dangling bits behind.

**Comment by DamianEdwards** [comment]
@dougbu this is simply removing lines that are now not required as the framework effectively calls them on your behalf in the cases that the template would have added them before, i.e. the AuthN/AuthZ middleware getting added automatically now if AuthN/AuthZ services are added.

**Review by DamianEdwards — APPROVED** [review]

**Comment by dougbu** [comment]
Got it @DamianEdwards, thanks. I'd resolve my comment if I could 😀

**Comment by captainsafia** [comment]
cc: @vlada-shubina @phenning for review on the changes related to accelerator keys for the "UseProgramMain" template option.

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json [hunk 1]
```diff
@@ -518,7 +518,7 @@
       "type": "parameter",
       "datatype": "bool",
       "defaultValue": "false",
-      "displayName": "Do not use top-level statements",
+      "displayName": "Do not use _top-level statements",
```
Comment (phenning) [inline src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:521]: should there be a corresponding change in templatestrings.en.json as well, there is one for Components-WebAssembly

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/localize/templatestrings.en.json [hunk 2]
```diff
@@ -36,11 +36,12 @@
   "symbols/CalledApiUrl/description": "URL of the API to call from the web app. This option only applies if --auth SingleOrg, --auth MultiOrg or --auth IndividualB2C without and ASP.NET Core host is specified.",
   "symbols/CallsMicrosoftGraph/description": "Specifies if the web app calls Microsoft Graph. This option only applies if --auth SingleOrg or --auth MultiOrg is specified.",
   "symbols/CalledApiScopes/description": "Scopes to request to call the API from the web app. This option only applies if --auth SingleOrg, --auth MultiOrg or --auth IndividualB2C without and ASP.NET Core host is specified.",
-  "symbols/UseProgramMain/displayName": "Do not use top-level statements",
+  "symbols/UseProgramMain/displayName": "Do not use _top-level statements",
+  "_symbols/UseProgramMain/displayName.comment": "Use '_' as accelerator key when translating.",
```
Comment (phenning) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/localize/templatestrings.en.json:40]: I don't se a corresponding template.json change in this template.

File: src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Program.Main.cs [hunk 3]
```diff
@@ -140,11 +140,6 @@ public static void Main(string[] args)
 
         app.UseRouting();
```
Comment (phenning) [inline src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Program.Main.cs:142]: There seem to be missing top-level statement related changes for RazorPages/StarterWeb and WebAPI

**Review by phenning — CHANGES_REQUESTED** [review]

File: src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json (hunk 1 above)
Comment (captainsafia) [reply src/ProjectTemplates/Web.ProjectTemplates/content/BlazorServerWeb-CSharp/.template.config/template.json:521]: 😲 Updated the remaining templates here. 🤦🏽‍♀️

**Review by brunolins16 — APPROVED** [review]
The template (`cs` files) changes look good for me; however, I don't know enough to review if the `json` changes are correct or not :)

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/localize/templatestrings.en.json [hunk 4]
```diff
@@ -36,7 +36,7 @@
   "symbols/CalledApiUrl/description": "URL of the API to call from the web app. This option only applies if --auth SingleOrg, --auth MultiOrg or --auth IndividualB2C without and ASP.NET Core host is specified.",
   "symbols/CallsMicrosoftGraph/description": "Specifies if the web app calls Microsoft Graph. This option only applies if --auth SingleOrg or --auth MultiOrg is specified.",
   "symbols/CalledApiScopes/description": "Scopes to request to call the API from the web app. This option only applies if --auth SingleOrg, --auth MultiOrg or --auth IndividualB2C without and ASP.NET Core host is specified.",
-  "symbols/UseProgramMain/displayName": "Do not use _top-level statements",
+  "symbols/UseProgramMain/displayName": "Do not _use _top-level statements",
```
Comment (DamianEdwards) [inline src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/localize/templatestrings.en.json:39]: Somehow ended up with two underscores here, I don't think that's intentional?

**Review by DamianEdwards — CHANGES_REQUESTED** [review]

File: src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/localize/templatestrings.en.json (hunk 4 above)
Comment (captainsafia) [reply src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/.template.config/localize/templatestrings.en.json:39]:
```suggestion
  "symbols/UseProgramMain/displayName": "Do not use _top-level statements",
```

**Review by DamianEdwards — APPROVED** [review]

**Review by vlada-shubina — APPROVED** [review]
Change related to accelerator key looks good to me

**Comment by captainsafia** [comment]
@phenning Do you mind giving this a green light if you're fine with the changes made?

**Review by phenning — APPROVED** [review]

**Comment by vijayrkn** [comment]
@deepchoudhery @zahalzel - Tagging so that you can make the related changes to the ASP.NET Core Scaffolding & MS Identity scaffolding templates.

**Comment by deepchoudhery** [comment]
@vijayrkn cool already been on it.

**Pull request merged**